components/SerialServer/src/serial.c

```diff
@@ -393,23 +406,27 @@ static void handle_char(uint8_t c)
             last_out = -1;
             printf(COLOR_RESET "\r\n --- SerialServer help ---"
                    "\r\n Escape char: %c"
-                   "\r\n 0 - %-2d switches input to that client"
+                   "\r\n # -    switches to client #"
                    "\r\n ?      shows this help"
                    "\r\n m      simultaneous multi-client input"
                    "\r\n d      switch between debugging modes"
                    "\r\n          0: no debugging"
                    "\r\n          1: debug multi-input mode output coalescing"
                    "\r\n          2: debug flush_buffer_line"
-                   "\r\n", ESCAPE_CHAR, num_getchar_clients - 1);
+                   "\r\n", ESCAPE_CHAR);
             statemachine = 1;
             break;
         default:
             last_out = -1;
             statemachine = 1;
-            if (c >= '0' && c < '0' + num_getchar_clients) {
+            if (c >= '0' && c <= '9') {
                 int client = c - '0';
```

Comment:
This should become `int badge = c - '0';` now, because we are selecting by badge ID.